core/src/main/java/com/expedia/www/haystack/client/LogData.java

```diff
@@ -0,0 +1,61 @@
+package com.expedia.www.haystack.client;
+
+import java.util.Collections;
+import java.util.Map;
+
+public class LogData {
+    private final Long timestamp;
+    private final Map<String, ?> fields;
+    private final String message;
+    private final Object payload;
+
+    public LogData(Long timestamp, String message) {
+        this.timestamp = timestamp;
+        this.message = message;
+        this.payload = null;
+        this.fields = null;
```

Comment:
I personally prefer a single constructor that sets all instance fields (four in your case) with overloaded constructors with the different signatures. Less code, but YMMV so if you like it better setting all fields in each constructor that's OK.

Reply:
All four aren't actually able to be set at once.  I will refactor one of them to remove some of the duplication.